decide/voting/tests.py

```diff
@@ -591,3 +618,57 @@ def createCensusEmptyError(self):
             self.cleaner.current_url
             == self.live_server_url + "/admin/voting/question/add/"
         )
+
+
+class BooleanQuestionTestCase(BaseTestCase):
+    def setUp(self):
+        self.question = Question.objects.create(
+            question_type="B", desc="What is your favorite color?", voteBlank=False
+        )
+        self.question.save()
+
+    def test_save_boolean_question(self):
+        self.assertEqual(QuestionOption.objects.count(), 2)
+        self.assertEqual(QuestionOption.objects.first().option, "Sí")
+        self.assertEqual(QuestionOption.objects.last().option, "No")
+
+    def test_str_representation(self):
+        self.assertEqual(str(self.question), "What is your favorite color?")
+
+    def test_one_option_boolean_question(self):
+        self.question = Question.objects.create(
+            question_type="B", desc="What is your favorite color?", voteBlank=False
+        )
+        self.question.save()
+
+        opt = QuestionOption(question=self.question, option="option {}".format(1))
+        with self.assertRaises(BadRequest):
+            opt.save()
```

Comment:
why would you save a question option after a bad request? I don't understand this, could you further explain?

Reply:
This method is used to verify that a specific exception (in this case, a BadRequest exception) is raised when a certain block of code is executed. The intention here is to check whether calling the save() method on the opt object raises a BadRequest exception. If the save() method does not raise a BadRequest exception, the test will fail.